### PR TITLE
Using realm database in android

### LINKED ISSUE #3158: Using Realm Database as an Alternative to SQLite and Coredata in Android
### Introduction paragraph (2-3 paragraphs):
Realm mobile database is a lightweight database that can be an alternative to both SQLite and Coredata in your Android projects. It is open source and developer-friendly.

Realm is faster and has lots of modern features, such as JSON support, a fluent API, data change notifications, cross-platform, fast querying, and  fully encrypted all of which make life easier for Android developers.

In this tutorial, we are going to learn the basics of Realm for Android.

### Key takeaways: 
- What is Realm mobile database.
- Why use Realm.
- How is it different from other Mobile Database.
- Getting started with Realm in Android

### References: 
- https://realm.io/

### PR DESCRIPTION
**Pull request by JoelKanyi**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3158

### CODE REVIEW

**Comment by hectorkambow** [comment]
@JoelKanyi 

[Using realm database as an alternative to sqlite and coredata in android #3425.pdf](https://github.com/section-engineering-education/engineering-education/files/7027943/Using.realm.database.as.an.alternative.to.sqlite.and.coredata.in.android.3425.pdf)


Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by Ericgacoki** [comment]
Greetings @JoelKanyi, Please use the latest version of your article to fix plagiarism issues and ping me when you're done.

**Comment by JoelKanyi** [comment]
@Ericgacoki I have done everything I could to remove the small plagiarism that was there.

**Comment by Ericgacoki** [comment]
@hectorkambow, please run a second plagiarism check on this article. Thanks.

**Comment by WanjaMIKE** [comment]
Some minor sentences seem to raise plagiarism. Could you check it out. 

[Using realm database as an alternative to sqlite and coredata in android #3425.pdf](https://github.com/section-engineering-education/engineering-education/files/7074718/Using.realm.database.as.an.alternative.to.sqlite.and.coredata.in.android.3425.pdf)

**Review by Ericgacoki — APPROVED** [review]

File: content/articles/using-realm-database-as-an-alternative-to-sqlite-and-coredata-in-android/index.md [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+import io.realm.annotations.RealmClass
+import io.realm.annotations.Required
+
+@RealmClass
+open class Note : RealmModel {
+    @PrimaryKey
+    var id: String = ""/engineering-education/using-realm-database-as-an-alternative-to-sqlite-and-coredata-in-android/
+
+    @Required
+    var description: String? = ""
```
Comment (Ericgacoki) [inline content/articles/using-realm-database-as-an-alternative-to-sqlite-and-coredata-in-android/index.md:165]: @JoelKanyi, this seems to have been added accidentally. I'll fix it ASAP.

**Comment by LinusMuema** [comment]
Hi @JoelKanyi 
Thanks for the amazing article. It is easy to follow through and quite informative.
Keep up the amazing job! :rocket: 

cc: @Ericgacoki

**Pull request merged**